specification/batch/Azure.Batch/models.tsp

```diff
@@ -0,0 +1,3929 @@
+
+  @doc("The settings for an authentication token that the Task can use to perform Batch service operations. If this property is set, the Batch service provides the Task with an authentication token which can be used to authenticate Batch service operations without requiring an Account access key. The token is provided via the AZ_BATCH_AUTHENTICATION_TOKEN environment variable. The operations that the Task can carry out using the token depend on the settings. For example, a Task can request Job permissions in order to add other Tasks to the Job, or check the status of the Job or of other Tasks under the Job.")
+  authenticationTokenSettings?: AuthenticationTokenSettings;
+}
+
+@doc("Specifies how the Batch service should respond when the Task completes.")
+model ExitConditions {
+  @doc("A list of individual Task exit codes and how the Batch service should respond to them.")
+  exitCodes?: ExitCodeMapping[];
+
+  @doc("A list of Task exit code ranges and how the Batch service should respond to them.")
+  exitCodeRanges?: ExitCodeRangeMapping[];
+
+  @doc("How the Batch service should respond if the Task fails to start due to an error.")
+  preProcessingError?: ExitConfig;
+
+  @doc("How the Batch service should respond if a file upload error occurs. If the Task exited with an exit code that was specified via exitCodes or exitCodeRanges, and then encountered a file upload error, then the action specified by the exit code takes precedence.")
+  fileUploadError?: ExitConfig;
+
+  @doc("How the Batch service should respond if the Task fails with an exit condition not covered by any of the other properties. This value is used if the Task exits with any nonzero exit code not listed in the exitCodes or exitCodeRanges collection, with a pre-processing error if the preProcessingError property is not present, or with a file upload error if the fileUploadError property is not present. If you want non-default behavior on exit code 0, you must list it explicitly using the exitCodes or exitCodeRanges collection.")
+  default?: ExitConfig;
+}
+
+@doc("""
+How the Batch service should respond if a Task exits with a particular exit
+code.
+""")
+model ExitCodeMapping {
+  @doc("A process exit code.")
+  code: int32;
+
+  @doc("How the Batch service should respond if the Task exits with this exit code.")
+  exitOptions: ExitConfig;
+}
+
+@doc("Specifies how the Batch service responds to a particular exit condition.")
+model ExitConfig {
+  @doc("An action to take on the Job containing the Task, if the Task completes with the given exit condition and the Job's onTaskFailed property is 'performExitConfigJobAction'. The default is none for exit code 0 and terminate for all other exit conditions. If the Job's onTaskFailed property is noaction, then specifying this property returns an error and the add Task request fails with an invalid property value error; if you are calling the REST API directly, the HTTP status code is 400 (Bad Request).")
```

Comment:
This highlights an issue with using recommended SDK naming conventions over the actual name in the API. 'performExitConfigJobAction' is not the enum value. It's 'performExitOptionsJobAction'. Because this doc string is used for both SDKs and REST API docs, this is now inaccurate.

Reply:
I think this is only really an issue with exitOptions, since the name is encoded in the enum value. I don't see a real problem with switching uploadOptions to uploadConfig, but if we do that we should add `@projectedName("client", "uploadConfig")` to the property as well.

Reply:
After thinking more about this, this seems like a general conflict between doc strings and projected names. We should investigate if there's any way can use string interpolation to reference the effective name of properties, models and enum values. Just in this one doc string there's another reference to the `onTaskFailed` property and the `noaction` enum value. We have these references all over the place, and the more renames we have, the more we're going to hit this issue.

Reply:
changed